packages/components/src/Grid/GridCell/GridCell.tsx

```diff
@@ -0,0 +1,54 @@
+import React, { PropsWithChildren } from "react";
+import styles from "./GridCell.css";
+
+/**
+ * Breakpoints that we support
+ */
+const breakpoints = ["xs", "sm", "md", "lg", "xl"] as const;
+
+/**
+ *  Grab the type from breakpoints const
+ */
+type ValuesOf<T extends readonly unknown[]> = T[number];
+type Breakpoints = ValuesOf<typeof breakpoints>;
+
+type Sizes = 1 | 2 | 3 | 4 | 5 | 6 | 7 | 8 | 9 | 10 | 11 | 12;
+
+export interface GridCellProps {
+  /**
+   * Set how much column width a cell takes from the grid
+   */
+  readonly size: { [Breakpoint in Breakpoints]?: Sizes };
+}
+
+const MAX_COLUMNS = 12;
+export const GRID_CELL_TEST_ID = "ATL-Grid-GridCell";
+
+export function GridCell({ size, children }: PropsWithChildren<GridCellProps>) {
+  const cssVariables = breakpoints.reduce(getCSSVariables(size), {});
```

Comment:
Is there a name more specific for this const that would help explain what CSS Variables are getting defined here)?
Maybe `cssSizeVariables`?
Also, consider a comment for this line, e.g. `// Build column sizes for all possible breakpoints to size column`? 
I just find this (and most reducers) easier to parse if I know what/why they are doing what they do. :)

Reply:
> I just find this (and most reducers) easier to parse if I know what/why they are doing what they do. :)

That's fair. Reducers is pretty cool, but they are confusing sometimes